move-mutation-test/src/cli.rs

```diff
@@ -0,0 +1,204 @@
+use move_model::metadata::LanguageVersion;
+use move_mutator::cli::{FunctionFilter, ModuleFilter};
+use move_package::CompilerConfig;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+/// Command line options for mutation test tool.
+#[derive(Parser, Default, Debug, Clone, Deserialize, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct CLIOptions {
+    /// Work only over specified modules.
+    #[clap(long, value_parser, default_value = "all")]
+    pub mutate_modules: ModuleFilter,
+
+    /// Work only over specified functions.
+    #[clap(short = 'f', long, value_parser, default_value = "all")]
+    pub mutate_functions: FunctionFilter,
+
+    /// Optional configuration file for mutator tool.
+    #[clap(long, value_parser)]
+    pub mutator_conf: Option<PathBuf>,
+
+    /// Save report to a JSON file.
+    #[clap(short, long, value_parser)]
+    pub output: Option<PathBuf>,
+
+    /// Use previously generated mutants.
+    #[clap(long, short, value_parser)]
+    pub use_generated_mutants: Option<PathBuf>,
+}
+
+/// This function creates a mutator CLI options from the given mutation-test options.
+#[must_use]
+pub fn create_mutator_options(options: &CLIOptions) -> move_mutator::cli::CLIOptions {
+    move_mutator::cli::CLIOptions {
+        mutate_functions: options.mutate_functions.clone(),
+        mutate_modules: options.mutate_modules.clone(),
+        configuration_file: options.mutator_conf.clone(),
+        // To run tests, compilation must succeed
+        verify_mutants: true,
+        ..Default::default()
+    }
+}
+
+/// This function checks if the mutator output path is provided in the configuration file.
+/// We don't need to check if the mutator output path is provided in the options as they were created
+/// from the mutation-test options which does not allow setting it.
+#[must_use]
+pub fn check_mutator_output_path(options: &move_mutator::cli::CLIOptions) -> Option<PathBuf> {
+    options
+        .configuration_file
+        .as_ref()
+        .and_then(|conf| move_mutator::configuration::Configuration::from_file(conf).ok())
+        .and_then(|c| c.project.out_mutant_dir)
+}
+
+/// The configuration options for running the tests.
+// Info: this set struct is based on TestPackage in `aptos-core/crates/aptos/src/move_tool/mod.rs`.
+#[derive(Parser, Debug)]
+pub struct TestBuildConfig {
+    /// Options for compiling a move package dir.
+    // We might move some options out and have our own option struct here - not all options are
+    // needed for mutation testing.
+    #[clap(flatten)]
+    pub move_pkg: MovePackageDir,
+
+    /// Dump storage state on failure.
+    #[clap(long = "dump")]
+    pub dump_state: bool,
+
+    /// A filter string to determine which unit tests to run.
+    #[clap(long, short)]
+    pub filter: Option<String>,
+
+    /// A boolean value to skip warnings.
+    #[clap(long)]
+    pub ignore_compile_warnings: bool,
+    // TODO: Unused in aptos-core:
```

Comment:
Interesting, does this flag do nothing in aptos-core, and that is what this TODO is suggesting to fix (in aptos-core)?

Reply:
The TODO comment was more a comment for myself: check what to do here, should we keep the argument or should we remove it.
And I think I'll remove it for now. 

ps. I just checked aptos-core by grepping this and saw that it seems unused:
```
aptos-core git:(main) ✗ rgrep instruction_execution_bound
vim crates/aptos/src/test/mod.rs +869               instruction_execution_bound: 100_000,
vim crates/aptos/src/move_tool/mod.rs +510          pub instruction_execution_bound: u64,
```

We could remove it from `aptos-core` or maybe add a TODO there - it seems like it's a leftover from some deprictaed code that was removed long time ago.